rpcs3/util/asm.hpp

```diff
@@ -138,10 +144,10 @@ namespace utils
 			return;
 		}
 
-#ifdef _M_X64
+#if defined(_M_X64)
 		return _m_prefetchw(ptr);
 #else
-		return __builtin_prefetch(ptr, 1, 0);
+		return __builtin_prefetch(ptr, 1, _MM_HINT_T0);
```

Comment:
why is this changed from 0 to _MM_HINT_T0, which is 1 ?

Reply:
Here I was not sure, but `_m_prefetchw` is defined like this in GCC:

```
extern __inline void __attribute__((__gnu_inline__, __always_inline__, __artificial__))
_m_prefetchw (void *__P)
{
  __builtin_prefetch (__P, 1, 3 /* _MM_HINT_T0 */);
}
```

So `_MM_HINT_T0` seemed correct.

Reply:
If this seems to show differences accross compilers why not just leave em all as they were with integers?

Reply:
![image](https://github.com/RPCS3/rpcs3/assets/23019877/9cb17c84-4f1f-43c0-b6a9-98828883ff8c)

Reply:
I prefer named constants to magic numbers and the fact that they are different depending on the platform speaks for constants IMO. I'd rather not revert it unless you insist.

Reply:
Don't change the meaning of the code. If you want named constants, make sure they're uniform across all compilers.
It makes more sense to leave this out and handle it in a separate PR. The changes in this PR are very low priority afterall.

Reply:
First of all, the mm hints are only meant for _mm_prefetch.
So _builtin_prefetch doesn't work with these constants and only uses numbers as far as I can tell looking at the documentation. Those numbers also don't seem to have the same meaning.
And then you also change the value 3 to _MM_HINT_T0 above, which doesn't even match.

So in my opinion you should just keep it as it was before, since you are not supposed to change the existing behavior in this pull request, which is solely meant to add another compiler.

Same goes for the Simd stuff you removed from existing compilers.

Reply:
Yup, this PR should be a CICD/build fix only. 0 semantic changes should be included or it will be impossible to diagnose any issues that creep in.

Reply:
Reverted. I get it now.